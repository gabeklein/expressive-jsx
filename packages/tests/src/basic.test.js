const test = require("./_adapter");

test("component", `
  () => do {
    <div>Hello World</div>
  }
`);

test("multiple", `
  () => do {
    <this>
      <div>Foo</div>
      <div>Bar</div>
    </this>
  }
`);

test("statement pass-thru", `
  () => do {
    const [x, setX] = useState();

    useEffect(() => {
      void x;
      setX();
    }, []);
    
    <div />
  }
`)