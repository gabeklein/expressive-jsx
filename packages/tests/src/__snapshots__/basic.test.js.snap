// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component 1`] = `"() => create(\\"div\\", {}, \\"Hello World\\");"`;

exports[`component 2`] = `"() => <div>Hello World</div>;"`;

exports[`multiple 1`] = `"() => create(Fragment, {}, create(\\"div\\", {}, \\"Foo\\"), create(\\"div\\", {}, \\"Bar\\"));"`;

exports[`multiple 2`] = `"() => <Fragment><div>Foo</div><div>Bar</div></Fragment>;"`;

exports[`statement pass-thru 1`] = `
"() => {
  const [x, setX] = useState();
  useEffect(() => {
    void x;
    setX();
  }, []);
  return create(\\"div\\", {});
};"
`;

exports[`statement pass-thru 2`] = `
"() => {
  const [x, setX] = useState();
  useEffect(() => {
    void x;
    setX();
  }, []);
  return <div />;
};"
`;
